packages/mcp-adapter/src/next/server-response-adapter.ts

```diff
@@ -57,7 +58,7 @@ export function createServerResponseAdapter(
         throw new Error('Buffer not supported');
       }
       if (!wroteHead) {
-        writeHead(200);
+        writeHead(statusCode, headers);
```

Comment:
This is the main bug in the code, if `writeHead` had not been called we returned 200 regardless of the `statusCode` value.